core/sql/executor/ExUdr.cpp

```diff
@@ -809,20 +809,17 @@ Int32 ExUdrTcb::fixup()
           else
             libOrJarName = myTdb().getContainerName();
           Int32 err = 0;
-          if(ComGenerateUdrCachedLibName(libOrJarName.data(),
+          if(err=ComGenerateUdrCachedLibName(libOrJarName.data(),
                                          myTdb().getLibraryRedefTime(),
                                          myTdb().getLibrarySchName(),
                                          dummyUser,
                                          cachedLibName, cachedLibPath))
             {
-              NAString cachedFullName = cachedLibPath+"/"+cachedLibName;
+             
                char errString[200];
                NAString errNAString;
-               sprintf(errString , "Error %d creating directory :",err); 
-               errNAString = errString;
-               errNAString += cachedFullName;
-             
-              
+               sprintf(errString , "Error %d creating directory %s :",err, cachedLibPath.data()); 
+               errNAString = errString;            
```

Comment:
Can cachedLibPath be longer than 200 characters?  Will errString size be long enough?

Reply:
Thanks for pointing this out. Fixing in both places this is called.